lib/report_generator.rb

```diff
@@ -10,8 +10,12 @@ def generate(query_id, query_params, recipients)
       query = DiscourseDataExplorer::Query.find(query_id)
       return [] unless query
 
+      creator = User.find(@creator_user_id)
+      return [] unless Guardian.new(creator).can_send_private_messages?
+
+      return [] if recipients.empty?
```

Comment:
```suggestion
      return [] if recipients.empty?

      creator = User.find(@creator_user_id)
      return [] unless Guardian.new(creator).can_send_private_messages?
```

Slightly more efficient since we don't run queries unnecessarily when `recipients` is empty.